### PR TITLE
gh-117478: Add `@support.requires_gil_enabled` decorator

### PR DESCRIPTION
**Pull request by colesbury**
* Issue: gh-117478

### CODE REVIEW

File: Lib/test/support/__init__.py [hunk 1]
```diff
@@ -837,6 +837,10 @@ def check_cflags_pgo():
 
 
 Py_GIL_DISABLED = bool(sysconfig.get_config_var('Py_GIL_DISABLED'))
+
+def requires_gil_enabled(msg="needs the GIL enabled"):
+    return unittest.skipIf(Py_GIL_DISABLED, msg)
```
Comment (colesbury) [inline Lib/test/support/__init__.py:843]: @Eclips4, I went with a slightly different implementation so that the callers can override the skip message.

**Review by Eclips4 — APPROVED** [review]
One nit: do not forget to put the new helper in ``support/__init__.__all__``, I think this should be a public API of support package.
Otherwise LGTM!

**Comment by Eclips4** [comment]
Oh, it would be nice if we add a few lines to the `Doc/lib/test.rst` (likewise for other `requires_X` helpers)

File: Doc/library/test.rst [hunk 2]
```diff
@@ -731,6 +731,11 @@ The :mod:`test.support` module defines the following functions:
    macOS version is less than the minimum, the test is skipped.
 
 
+.. decorator:: requires_gil_enabled
+
+   Decorator for skipping tests on the free-threaded build.
```
Comment (Eclips4) [inline Doc/library/test.rst]:
```suggestion
   Decorator for skipping tests on the free-threaded build. If the :term:`GIL` is disabled, the test is skipped.
```

**Pull request merged**